File: OrderSystem_BE/src/main/java/com/example/orderSystem/domain/order/Order.java
package com.example.orderSystem.domain.order;

import com.example.orderSystem.api.order.dto.OrderCreateRequestDto;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Getter;

@Getter @Builder @AllArgsConstructor
public class Order {

    private Long id;
    private String name;
    private Status status;
    private int quantity;


    public static Order of(OrderCreateRequestDto dto) {
        return Order.builder()
                .name(dto.getName())
                .quantity(dto.getQuantity())
                .status(Status.RECEIVED)
                .build();
    }
}

File: OrderSystem_FE/src/types/order.ts
export interface OrderCreateRequestDto{
    name: string;
    quantity: number;
}

export interface Order{
    id: number;
    name: string;
    status: Status;
    quantity: number;
}

export enum Status{
    RECEIVED="접수됨",
    PROCESSING="처리 중",
    FINISHED="처리 완료"
}